### DISCUSSION TITLE
What is the difference between OnModel* and OnRecord* event hooks?

### DISCUSSION THREAD

**Discussion by joetifa2003** [pocketbase/pocketbase#5264] [Q&A]
What i want to do is to ingest data from multiple collections to search index, when to use which of these hooks when using pocketbase as a framework?

And why is e.Model an interface in OnModel* hooks?

**Comment by ganigeorgiev** [comment]
_See https://github.com/pocketbase/pocketbase/discussions/4966_.

The `*Request` suffixed hooks fire only on API request, while the `OnModel*` fire on each db model action (a single request can trigger one or more db changes).

The `OnModel*` hooks fire for any struct that satisfy the `core.Model` interface, it could be `models.Record`, `models.Collection`, `models.Admin`, custom models, etc.

With the [refactoring](https://github.com/pocketbase/pocketbase/discussions/4355) for convenience and to avoid the type assertion there are "projection/proxy" hook equivalent to the specific Model representation (e.g. `OnRecordCreate`, `OnCollectionCreate`).
